### PR TITLE
Enviroments rework

### PR DESCRIPTION
**Pull request by Zekvyrin**
Re-worked the enviroments, based on rtr project.

(and allow usage of a `.env` file which is to be ignored)

### CODE REVIEW

File: .env.example [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# The current myRotary environment.
```
Comment (JohnAlbin) [inline .env.example]: Remove references to clients please.

File: gatsby-config.js [hunk 2]
```diff
@@ -6,8 +6,10 @@
 
 // Read in environment variables based on the current `NODE_ENV`.
 require('dotenv').config({
-  path: `.env.${process.env.NODE_ENV}`,
+  path: `.environments/${process.env.NODE_ENV || 'local'}.env`,
 });
+require('dotenv').config({ path: `.env` });
+require('dotenv').config({ path: `.env.example` });
```
Comment (JohnAlbin) [inline gatsby-config.js]: If the later calls to `.config()` override earlier calls, why is the `.env.sample` called last?

Comment (Zekvyrin) [reply gatsby-config.js]: I've tested it, and the "priority" it goes top to bottom.

I guess it's confusing but this is how it works.

Comment (JohnAlbin) [reply gatsby-config.js]: Ah! I see.

```
We will never modify any environment variables that have already been set. In particular, if there is a variable in your .env file which collides with one that already exists in your environment, then that variable will be skipped.
```

File: .env.example [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# The current environment.
+# Used for conditional loading of .env files.
+# Options:
+# - local: local development
+# - lagoon: lagoon feature branch environment
+# - dev: dev on xxx
+# - prod: prod on xxx
+NODE_ENV=local
```
Comment (pmelab) [inline .env.example]: @JohnAlbin @Zekvyrin : Is this a problem with actual `NODE_ENV` uses? Some systems (also gatsby) populate `NODE_ENV` themselves to trigger optimizations. `gatsby build` for example sets `NODE_ENV` to `production`. I suspect that we effectively disable a couple of optimizations by setting it to `prod`.

Comment (JohnAlbin) [reply .env.example]: I just tested `yarn build` and `yarn develop` and Gatsby completely overrides NODE_ENV when it's run. So the change to NODE_ENV in `.env.example` won't do anything. We should remove it.

Comment (JohnAlbin) [reply .env.example]: @Zekvyrin said this came from the R***** project, but looking at that code, I can see that it is using MRX_ENV variable and not NODE_ENV to switch API endpoints. That needs to be changed here as re-using NODE_ENV isn't going to work the way we want.

Comment (JohnAlbin) [reply .env.example]: I reverted this commit and we can discuss in the new PR. https://github.com/AmazeeLabs/gatsby-starter-amazee/pull/30

**Pull request merged**